doc/source/dev/contributor/quickstart_ubuntu.rst

```diff
@@ -0,0 +1,113 @@
+:orphan:
+
+.. _quickstart-ubuntu:
+
+=================================================
+Development Environment Quickstart Guide (Ubuntu)
+=================================================
+
+This quickstart guide will cover:
+
+* setting up and maintaining a development environment, including installing compilers and SciPy build dependencies;
+* creating a personal fork of the SciPy repository on GitHub;
+* using git to manage a local repository with development branches;
+* performing an in-place build of SciPy; and
+* creating a virtual environment that adds this development version of SciPy to the Python path
+
+in Ubuntu.
+
+.. note::
+
+	This guide does not present the only way to set up a development environment; there are many valid choices of Python distribution, C/Fortran compiler, and installation options. The steps here can often be adapted for other choices, but we cannot provide documentation tailored for them.
+
+	This guide assumes that you are starting without an existing Python 3 installation. If you already have Python 3, you might want to uninstall it first to avoid ambiguity over which Python version is being used at the command line.
+
+.. _quickstart-ubuntu-build:
+
+Building SciPy
+--------------
+
+*Consider watching the companion video* `Anaconda SciPy Dev: Part I (macOS)`_ *before starting. Even though the video is intended for macOS users, most of the procedures are the same or very similar.*
+
+#. Download, install, and test the latest release of the `Anaconda Distribution of Python`_. In addition to the latest version of Python 3, the Anaconda distribution includes dozens of the most popular Python packages for scientific computing, the Spyder integrated development environment (IDE), the ``conda`` package manager, and tools for managing virtual environments.
```

Comment:
Should we be documenting Anaconda by default?

Anaconda includes quite a few packages not used in building SciPy and leads to a reasonable large download, which may be a barrier to users with slow internet connections. See this [NumPy mailing list](http://numpy-discussion.10968.n7.nabble.com/how-much-does-binary-size-matter-td46885.html#a46886) discussion. Note that they are discussing a **16mb** download. The Anaconda installer on its own is ~650mb.

Second, when using the ``apt`` machinery, a user who can't (or doesn't want to) use Anaconda must specify the requires packages individually. In that case having a reference of the required packages will be useful (plus whenever I rebuild SciPy I can simply copy and paste the requirements)

Reply:
1. This tutorial is for the average beginning developer. Since the smallest hard drive one can easily buy these days is hundreds of GB and sufficiently large USB flash drives are frequently given away for free, I don't expect the [5GB requirement ](https://docs.anaconda.com/anaconda/install/) to be a problem for the average beginning developer. I'm not saying that this is not a problem for anyone on any platform - I've had a hard time staying under a 2GB quota on a remote server I wanted to use for SciPy development - but that Anaconda serves a wide audience. Moreover, in situations where space is at a premium, these instructions are easily adapted for use with [Miniconda](https://docs.conda.io/en/latest/miniconda.html), which comes with `conda` but without all the other packages. I would be happy to review such a PR, or a PR for any other choice of Python distribution, package manager, virtual environment system, and editor. (The choice of all those is easy if you pick Anaconda, BTW.) At some point we can combine all these to replace the [current build documentation ](https://docs.scipy.org/doc/scipy/reference/building/index.html) entirely, which has never worked for me without modification and (hopefully most would agree) not as easy for a first-time developer to follow as the new guides and videos.

2. Like Step 6 of [Installing SciPy](https://github.com/scipy/scipy/blob/d24db98f364c2a0e498c537ca25d59781f2ad6ed/doc/source/dev/contributor/quickstart_ubuntu.rst#installing-scipy) or do you mean something else?